File: Source/Padutronics.Windows.Win32.Api.CWrapper/D3D11/ID3D11DeviceContext.cpp
#include "ID3D11DeviceContext.hpp"

void ID3D11DeviceContext_ClearRenderTargetView(ID3D11DeviceContext& _this, ID3D11RenderTargetView* pRenderTargetView, const FLOAT ColorRGBA[4])
{
    _this.ClearRenderTargetView(pRenderTargetView, ColorRGBA);
}

void ID3D11DeviceContext_OMSetRenderTargets(ID3D11DeviceContext& _this, UINT NumViews, ID3D11RenderTargetView* const* ppRenderTargetViews, ID3D11DepthStencilView* pDepthStencilView)
{
    _this.OMSetRenderTargets(NumViews, ppRenderTargetViews, pDepthStencilView);
}

void ID3D11DeviceContext_RSSetViewports(ID3D11DeviceContext& _this, UINT NumViewports, const D3D11_VIEWPORT* pViewports)
{
    _this.RSSetViewports(NumViewports, pViewports);
}

File: Source/Padutronics.Windows.Win32.Api/D3D11/ID3D11DeviceContext.cs
using System.Runtime.InteropServices;

namespace Padutronics.Windows.Win32.Api.D3D11;

[Guid("c0bfa96c-e089-44fb-8eaf-26f8796190da")]
public interface ID3D11DeviceContext : ID3D11DeviceChild
{
    void ClearRenderTargetView(ID3D11RenderTargetView pRenderTargetView, float[] ColorRGBA);
    void OMSetRenderTargets(int NumViews, ID3D11RenderTargetView[]? ppRenderTargetViews, ID3D11DepthStencilView? pDepthStencilView);
    void RSSetViewports(int NumViewports, D3D11_VIEWPORT[]? pViewports);
}